### PR TITLE
Removing Globals part 3 of 4

### PR DESCRIPTION
**Pull request by Neelesh99**
# Summary

Finished removing global variables, all rules information is contained entirely within RuleContext struct in gamestate or in client local caches.

## Test Plan

Had to modify many tests, all now passing.

### CODE REVIEW

File: internal/server/iigointernal/executive_test.go [hunk 1]
```diff
@@ -690,8 +690,10 @@ func TestReplyAllocationRequest(t *testing.T) {
 		},
 	}
 
-	rules.PullRuleIntoPlay("allocation_decision")
-	rules.PullRuleIntoPlay("check_allocation_rule")
+	rulesInPlay := map[string]rules.RuleMatrix{}
+
+	rules.PullRuleIntoPlayInternal("allocation_decision", generateRuleStore(), rulesInPlay)
```
Comment (lhl2617) [inline internal/server/iigointernal/executive_test.go]: even in a test these should be checked

File: internal/server/iigointernal/executive_test.go [hunk 2]
```diff
@@ -894,8 +910,8 @@ func TestBroadcastTaxation(t *testing.T) {
 		},
 	}
 
-	rules.PullRuleIntoPlay("tax_decision")
-	rules.PullRuleIntoPlay("check_taxation_rule")
+	rules.PullRuleIntoPlayInternal("tax_decision", generateRuleStore(), rulesInPlay)
```
Comment (lhl2617) [inline internal/server/iigointernal/executive_test.go]: +1

**Comment by lhl2617** [comment]
can you update from main? thx!

File: internal/common/baseclient/baseSpeaker.go [hunk 3]
```diff
@@ -1,16 +1,18 @@
 package baseclient
 
 import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
 	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
 	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
 )
 
 type BaseSpeaker struct {
+	gameState gamestate.ClientGameState
```
Comment (NGovani) [inline internal/common/baseclient/baseSpeaker.go]: doesnt this need to be initialised in GetClientSpeakerPointer() in baseclient iigo

**Review by NGovani — CHANGES_REQUESTED** [review]
one issue otherwise LGTM

File: internal/server/iigointernal/executive_test.go (hunk 1 above)
Comment (Neelesh99) [reply internal/server/iigointernal/executive_test.go]: Oooo nice catch, changed in latest commit

File: internal/server/iigointernal/executive_test.go (hunk 2 above)
Comment (Neelesh99) [reply internal/server/iigointernal/executive_test.go]: As above, thanks for finding it

File: internal/common/baseclient/baseSpeaker.go (hunk 3 above)
Comment (Neelesh99) [reply internal/common/baseclient/baseSpeaker.go]: Ah I knew I forgot smth, thanks, changed in latest commit

**Comment by Neelesh99** [comment]
> can you update from main? thx!

Updated

**Comment by lhl2617** [comment]
>  (no, it was never 4) 

what about monitoring? 
image

**Comment by lhl2617** [comment]
game still crashing after this fix.

**Comment by Neelesh99** [comment]
> > (no, it was never 4)
> 
> what about monitoring?
> image

Dammit, you're right. @preetl is working on it. I believe her computer had decided it wasn't going to work on it until about an hour ago.

**Comment by Neelesh99** [comment]
> game still crashing after this fix.

Hard crash in terminal or the NaN thing?

**Comment by lhl2617** [comment]
> > game still crashing after this fix.
> 
> Hard crash in terminal or the NaN thing?

 NaN

**Comment by Neelesh99** [comment]
> > > game still crashing after this fix.
> > 
> > 
> > Hard crash in terminal or the NaN thing?
> 
> NaN

Ok, hopefully monitoring is the culprit then.

**Comment by lhl2617** [comment]
ok. monitoring is not the culprit, but let's remove it as well. the problem is in the defualt president team 3.

**Comment by Neelesh99** [comment]
> ok. monitoring is not the culprit, but let's remove it as well. the problem is in the default president team 3.

Damn that Team 3, always messing with our stuff

**Review by NGovani — APPROVED** [review]
Lgtm, if it works in testing merge it in

**Pull request merged**